mm/2s2h/Rando/ActorBehavior/EnToto.cpp

```diff
@@ -38,4 +39,19 @@ void Rando::ActorBehavior::InitEnTotoBehavior() {
         Message_BombersNotebookQueueEvent(gPlayState, BOMBERS_NOTEBOOK_EVENT_MET_TOTO);
         Message_BombersNotebookQueueEvent(gPlayState, BOMBERS_NOTEBOOK_EVENT_MET_GORMAN);
     });
+
+    COND_VB_SHOULD(VB_TOTO_START_SOUND_CHECK, IS_RANDO && RANDO_SAVE_OPTIONS[RO_SHUFFLE_OCARINA_BUTTONS], {
+        EnToto* totoActor = va_arg(args, EnToto*);
+        if (totoActor->text->textId == 0x2B24) {
+            if (!(Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_HUMAN) ||
+                  Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_DEKU) ||
+                  Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_GORON) ||
+                  Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_ZORA))) {
```

Comment:
One problem with this approach is if the player can play at least one of the songs, but goes into the performance as a form for which they do not have the needed notes. The lockout will still happen. Since you cannot get the check without playing all parts anyway, maybe we just AND these instead and call it a day? This would also prevent the skip ballad enhancement from letting the player cheat through performances they cannot do.

```suggestion
            if (!(Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_HUMAN) &&
                  Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_DEKU) &&
                  Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_GORON) &&
                  Rando::Logic::canPlaySong(OCARINA_SONG_WIND_FISH_ZORA))) {
```